From 0515e6d39b60cbb6d494a4fbe6c84780718c1d62 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 2 Apr 2022 14:13:01 +0200
Subject: [PATCH] generator: allow /etc/locale.conf to not exist

musl-based Linux distributions, e.g. Alpine Linux, don't provide
this file and default to UTF-8. As such, don't error-out if it
doesn't exist and just assume UTF-8 support.
---
 generator/console.go | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/generator/console.go b/generator/console.go
index 65de063..c512e20 100644
--- a/generator/console.go
+++ b/generator/console.go
@@ -3,6 +3,7 @@ package main
 import (
 	"bytes"
 	"compress/gzip"
+	"errors"
 	"fmt"
 	"io"
 	"os"
@@ -24,14 +25,14 @@ func (img *Image) enableVirtualConsole(vConsolePath, localePath string) (*Virtua
 	// adding keymap
 	if keymap, ok := vprop["KEYMAP"]; ok {
 		lconf, err := os.ReadFile(localePath)
-		if err != nil {
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
 			return nil, err
 		}
 		lprop := parseProperties(string(lconf))
 
 		lang := lprop["LANG"]
 		debug("detected language - '%s'", lang)
-		conf.Utf = strings.HasSuffix(strings.ToLower(lang), "utf-8")
+		conf.Utf = lang == "" || strings.HasSuffix(strings.ToLower(lang), "utf-8")
 		conf.KeymapFile = "/console/keymap"
 
 		blob, err := loadKeymap(keymap, vprop["KEYMAP_TOGGLE"], conf.Utf)
